rust/index/bindings.cpp

```diff
@@ -155,11 +162,6 @@ class Index
         }
         AllowAndDisallowListFilterFunctor filter = AllowAndDisallowListFilterFunctor(allow_list, disallow_list);
         std::priority_queue<std::pair<dist_t, hnswlib::labeltype>> res = appr_alg->searchKnn(query_vector, k, &filter);
-        if (res.size() < k)
```

Comment:
was this untested behavior?

Reply:
Not really, this path was never hit, it should have been removed, we always set k to min(k, len()) in calling code.

Reply:
maybe it pays to be defensive here, I haven't looked but is there any chance of UB if we do send `k > res.size()` ?

Reply:
nope, hnsw handles it fine, it just underpopulates the results, this was carried over from the hnsw bindings for python that choose to explicitly throw. We are changing the contract from the python bindings that it is OK to return < K results